### DISCUSSION TITLE
Proposed feature: Default external program

### DISCUSSION THREAD

**Discussion by wd5gnr** [flameshot-org/flameshot#2809] [Ideas]
Have a look at this: https://github.com/wd5gnr/flameshot/tree/default-ext-app

Fairly big change and not bug-related so I didn't want to do a PR yet, but will.

Here's how it works.

1. By default, nothing has changed.
2. When you run an external program there is a check at the bottom to set the launched program/terminal status as default.
3. If you check that, next time you call the external program, this will run instead of the launcher widget.
4. If you change your mind, you can turn the external program off (and terminal status) in configuration. However, you can only set the program name through the launcher (so to change defaults, you have to turn off, run external program, set default again).

I also cleaned up some of the config code from the XYWH display stuff, so that should get merged even if consensus is this is undesirable. I also don't know if this works on Windows since it looks like there may be a different path for the external launch on Windows? Or else there is dead code at showOpenWithMenu() in openwithprogram.cpp. I couldn't find where this actually executes.

I originally thought about making it where right click opens default and left click stays as is and then having an option to simply flip that, but it wasn't clear to me how to catch the right click on buttons, but that is another possibility. The "set" code would stay the same as would the launching logic.

**Comment by mmahmoudian** [comment]
I remember we have had a feature request for this: #2091
